src/publisher_factory.cpp

```diff
@@ -35,101 +35,93 @@
 
 PublisherFactory::PublisherFactory() { m_currentMode = ModeTypes::NONE; };
 
-void PublisherFactory::createNew(ModeTypes mode, ros::NodeHandle nHandle,
-                                 const std::shared_ptr<aditof::Camera> &camera,
-                                 aditof::Frame **frame) {
+void PublisherFactory::createNew(
+  ModeTypes mode, ros::NodeHandle nHandle, const std::shared_ptr<aditof::Camera> & camera,
+  aditof::Frame ** frame)
+{
+  ros::Time timeStamp = ros::Time::now();
 
-    ros::Time timeStamp = ros::Time::now();
+  if (*frame != nullptr) (*frame)->~Frame();
```

Comment:
Can you call the destructor with delete keyword?
delete *frame;

Reply:
Is there a specific reason for it?

Reply:
I usually used just delete keyword to call the destructor. Calling explicity will call the base destructor or just the called destructor?

Reply:
I see. Since frame hase no child members, for this case we might keep this form